sarcophagus-org/sarcophagus-v2-app
# 69 duplicate archaeologists updating with pub key

# Archaeologist loading fixes

Fixes some issues involving archaeologists not loading correctly. 

### Notes
- Refactored a lot of the archaeologist node discovery. A lot of this had to be changed to be compatible with how React works, I tried to keep the underlying logic as close as possible to the original.
- Removed P2Pprovider and added the state it managed to the store
- Removed archaeologist feature, since it's only being used for the archaeologist page which we don't need anymore. 
- Moved the code in `discovery.ts` to the `useLibp2p.ts` hook. 
- Major changes were made to `useLoadArchaeologists.ts` and facilitates reading the archaeologist profiles from the contract. 
- The `Libp2pNode` is now kept in the store
- There are new actions `SetArchaeologistOnlineStatus` which triggers during the discovery process and `SetArchaeologistConnect`.
- The archaeologist list populates only once when the component mounts. The only need to update the list is if an archaeologist is registered, which we can listen for through a websocket if we needed to. 

### Testing 
- Run the sarcophagus contract locally (follow sarco contracts readme) 
- Run the arch service locally (follow the arch service readme). This will give you one arch to work with. 
- Navigate to the embalm portion of the web app (http://localhost:3000/embalm) and go to step 9: Select Archaeologists 
- Wait a few seconds and expect an archaeologist to appear in the list. If this takes more than 30 seconds then something is wrong. 

![image](https://user-images.githubusercontent.com/34484576/192049615-22a2cf6e-7e43-4570-a72c-71c556b6aee2.png)
